src/commissioner/Discoverer.cpp

```diff
@@ -0,0 +1,264 @@
+/*
+ *    Copyright (c) 2020-2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <commissioner/Discoverer.h>
+
+namespace chip {
+namespace Commissioner {
+namespace CommissionableNodeDiscoverer {
+
+#if CONFIG_NETWORK_LAYER_BLE
+BleDiscoverer::BleDiscoverer(Controller::DeviceControllerSystemState & systemState, Delegate ** delegate) :
```

Comment:
nit / optional: I recommend moving `BleDiscoverer` and `DnssdDiscoverer` into their own files.

Reply:
I have moved the implementations to their own cpp files.

The header I think is still pretty concise and I left that as a single file.